application/src/main/java/org/togetherjava/tjbot/commands/free/FreeHelpChannelsRoutine.java

```diff
@@ -0,0 +1,56 @@
+package org.togetherjava.tjbot.commands.free;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.utils.TimeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.Routine;
+
+import java.time.OffsetDateTime;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Routine that automatically marks inactive help channels as free.
+ * <p>
+ * The routine runs every 10 minutes, and checks for help channels that have been inactive for more
+ * than 2 hours.
+ * </p>
+ */
+public class FreeHelpChannelsRoutine implements Routine {
+    private final FreeChannelMonitor freeChannelMonitor;
+
+
+    /**
+     * Creates an instance of {@link FreeHelpChannelsRoutine}.
+     *
+     * @param freeChannelMonitor the {@link FreeChannelMonitor} to use for marking channels as free
+     */
+    public FreeHelpChannelsRoutine(@NotNull FreeChannelMonitor freeChannelMonitor) {
+        this.freeChannelMonitor = freeChannelMonitor;
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        jda.getTextChannels()
+            .stream()
+            .filter(channel -> freeChannelMonitor.isMonitoringChannel(channel.getIdLong()))
+            .filter(channel -> freeChannelMonitor.isChannelBusy(channel.getIdLong()))
```

Comment:
I don't know what your opinion is on adding overloads that take a TextChannel?
Would make the filters a lot more readable, and wouldn't add a lot more code.

Could also create a private method

Reply:
I am not sure what do you mean by adding overloads.

Reply:
You do not know what an overload is?

Imagine you having a print method:

```java
void print(String)
void print(Integer)
```

These are overloads, overloads are multiple methods with the same name.

Reply:
I know what's method overloading. I didn't understand what methods did you suggest to overload :)

Reply:
Should have made that clear the next time "what do you mean by adding overloads" sounds like you don't know what an overload is.
And see the message above, I'm explaining this case there

Reply:
Yea, I saw that.